src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java

```diff
@@ -113,31 +115,50 @@ public RandomTxPowerInitializer(
 		this.rng = rng;
 	}
 
-	/** Get a random tx power in [MIN_TX_POWER, MAX_TX_POWER], both inclusive */
-	public int getRandomTxPower() {
-		return rng.nextInt(TPC.MAX_TX_POWER + 1 - TPC.MIN_TX_POWER) +
-			TPC.MIN_TX_POWER;
-	}
-
 	@Override
 	public Map<String, Map<String, Integer>> computeTxPowerMap() {
-		int defaultTxPower = this.getRandomTxPower();
-		logger.info("Default power: {}", defaultTxPower);
+		Integer defaultTxPower = MAX_TX_POWER;
+		if (!setDifferentTxPowerPerAp) {
+			List<Integer> txPowerChoices =
+				new ArrayList<>(DEFAULT_TX_POWER_CHOICES);
+			for (String serialNumber : model.latestState.keySet()) {
+				for (String band : UCentralConstants.BANDS) {
+					txPowerChoices = updateTxPowerChoices(
+						band,
+						serialNumber,
+						txPowerChoices
+					);
+				}
+			}
+
+			defaultTxPower =
+				txPowerChoices.get(rng.nextInt(txPowerChoices.size()));
```

Comment:
doesn't this violate the choices available since the set of tx power choices won't necessarily be the same across APs due to reading user config? e.g. AP A returns [1, 2, 3] and B returns [4, 5, 6] (just an example). Depending on iteration order we'd be violating either A or B's requested power.

Reply:
Yes, you're right. When `setDifferentTxPowerPerAp` is false, `txPowerChoices` is computed as the shared tx powers across different APs, and then `dwfaultTxPower` is picked from that choices. If there's no shared tx powers (like you case), the default is used.